includes/CLI.php

```diff
@@ -132,19 +132,40 @@ public static function customers( $args, $assoc_args ) {
 
 		$time_start = microtime( true );
 
+		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating coupons', $amount );
+
 		Generator\Customer::disable_emails();
-		$progress = \WP_CLI\Utils\make_progress_bar( 'Generating customers', $amount );
-		for ( $i = 1; $i <= $amount; $i++ ) {
-			Generator\Customer::generate();
-			$progress->tick();
+
+		add_action(
+			'smoothgenerator_customer_generated',
+			function () use ( $progress ) {
+				$progress->tick();
+			}
+		);
+
+		$remaining_amount = $amount;
+		$generated        = 0;
+
+		while ( $remaining_amount > 0 ) {
+			$batch = $remaining_amount > Generator\Customer::MAX_BATCH_SIZE ? Generator\Customer::MAX_BATCH_SIZE : $remaining_amount;
```

Comment:
As in a related PR, we could optionally use PHP's `min()` function here (slightly shorter, possibly easier to read than a ternary).

Reply:
9e0bdbed7f8a101a2d4accc904673531a5870f42